src/artemis/artemis.service.ts

```diff
@@ -112,28 +156,50 @@ export class ArtemisService {
     });
   }
 
-  private async setupQueue(topicName: TopicName): Promise<QueueEntry> {
-    const entry = this.queueStore[topicName];
+  private async getConnection(): Promise<Connection> {
+    if (!this.connectionPromise) {
```

Comment:
I thought I _might_ have been getting a race, so added this kind of cache.

I don't think I was, but this way seems safer to make sure we only have one connection